pbrd/pbr_vty.c

```diff
@@ -322,27 +322,20 @@ DEFPY (pbr_policy,
 
 	if (no) {
 		if (strcmp(pbr_ifp->mapname, mapname) == 0) {
-			strcpy(pbr_ifp->mapname, "");
-
+			pbr_ifp->mapname[0] = '\0';
 			if (pbrm)
 				pbr_map_interface_delete(pbrm, ifp);
 		}
 	} else {
-		if (strcmp(pbr_ifp->mapname, "") == 0) {
-			strcpy(pbr_ifp->mapname, mapname);
-
-			if (pbrm)
-				pbr_map_add_interface(pbrm, ifp);
-		} else {
-			if (!(strcmp(pbr_ifp->mapname, mapname) == 0)) {
-				old_pbrm = pbrm_find(pbr_ifp->mapname);
-				if (old_pbrm)
-					pbr_map_interface_delete(old_pbrm, ifp);
-				strcpy(pbr_ifp->mapname, mapname);
-				if (pbrm)
-					pbr_map_add_interface(pbrm, ifp);
-			}
+		if (strcmp(pbr_ifp->mapname, "") != 0) {
+			old_pbrm = pbrm_find(pbr_ifp->mapname);
+			if (old_pbrm)
+				pbr_map_interface_delete(old_pbrm, ifp);
 		}
+		snprintf(pbr_ifp->mapname, sizeof(pbr_ifp->mapname),
```

Comment:
Don't you want to check for truncation as well? I think `strlcpy` fits better this case.

e.g. `if (strlcpy(pbr_ifp->mapname, mapname, sizeof(pbr_ifp->mapname)) >= sizeof(pbr_ifp->mapname))`

Reply:
```
RETURN VALUES
     Besides quibbles over the return type (size_t versus int) and signal han-
     dler safety (snprintf(3) is not entirely safe on some systems), the fol-
     lowing two	are equivalent:

	   n = strlcpy(dst, src, len);
	   n = snprintf(dst, len, "%s",	src);
```
I'm not worried about truncation here, this is just to remove the overflow